presto-base-arrow-flight/src/test/java/com/facebook/plugin/arrow/TestArrowFlightNativeQueries.java

```diff
@@ -0,0 +1,427 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.plugin.arrow;
+
+import com.facebook.airlift.log.Logger;
+import com.facebook.plugin.arrow.testingServer.TestingArrowProducer;
+import com.facebook.presto.sql.analyzer.FeaturesConfig;
+import com.facebook.presto.testing.QueryRunner;
+import com.facebook.presto.tests.AbstractTestQueryFramework;
+import com.facebook.presto.tests.DistributedQueryRunner;
+import com.google.common.collect.ImmutableMap;
+import org.apache.arrow.flight.FlightServer;
+import org.apache.arrow.flight.Location;
+import org.apache.arrow.memory.RootAllocator;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Ignore;
+import org.testng.annotations.Test;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.net.URI;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Map;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.function.BiFunction;
+
+import static com.facebook.plugin.arrow.ArrowFlightQueryRunner.getProperty;
+import static com.facebook.plugin.arrow.testingConnector.TestingArrowFlightPlugin.ARROW_FLIGHT_CATALOG;
+import static com.facebook.plugin.arrow.testingConnector.TestingArrowFlightPlugin.ARROW_FLIGHT_CONNECTOR;
+import static java.lang.String.format;
+import static org.testng.Assert.assertTrue;
+
+public class TestArrowFlightNativeQueries
+        extends AbstractTestQueryFramework
+{
+    private static final Logger log = Logger.get(TestArrowFlightNativeQueries.class);
+    private RootAllocator allocator;
+    private FlightServer server;
+    private DistributedQueryRunner arrowFlightQueryRunner;
+
+    @BeforeClass
+    public void setup()
+            throws Exception
+    {
+        arrowFlightQueryRunner = getDistributedQueryRunner();
+        //arrowFlightQueryRunner.getDefaultSession().getSystemProperty()
+
+        allocator = new RootAllocator(Long.MAX_VALUE);
+        Location location = Location.forGrpcTls("localhost", getServerPort());
+        File certChainFile = new File("src/test/resources/server.crt");
+        File privateKeyFile = new File("src/test/resources/server.key");
+
+        server = FlightServer.builder(allocator, location, new TestingArrowProducer(allocator))
+                .useTls(certChainFile, privateKeyFile)
+                .build();
+
+        server.start();
+        log.info("Server listening on port %s", server.getPort());
+    }
+
+    @AfterClass(alwaysRun = true)
+    public void close()
+            throws InterruptedException
+    {
+        arrowFlightQueryRunner.close();
+        server.close();
+        allocator.close();
+    }
+
+    public static int getServerPort()
+    {
+        return 9443;
```

Comment:
We should use PortUtil `velox/exec/tests/utils/PortUtil.h` to get a random port and avoid test failure due to the port being unavailable.

Reply:
PortUtil is only in C++, I updated with a Java equivalent here and will go back and try to use a random port with PortUtil in the native arrow connector tests

Reply:
both native and additional java tests are now using a random free port